reco_utils/recommender/deeprec/models/base_model.py

```diff
@@ -385,16 +358,16 @@ def load_model(self, model_path=None):
             raise IOError("Failed to find any matching files for {0}".format(act_path))
 
     def fit(self, train_file, valid_file, test_file=None):
-        """
-        Fit the model with train_file. Evaluate the model on valid_file per epoch to observe the training status.
+        """Fit the model with train_file. Evaluate the model on valid_file per epoch to observe the training status.
         If test_file is not None, evaluate it too.
+        
```

Comment:
This function accepting both validation and test data is strange to me... why does this function evaluate the model both on validation and test data?

Also, having validation data as an input to `fit` function doesn't sound like following single responsibility principle.
E.g. TF has a dedicated function for this ([train_and_evaluate](https://www.tensorflow.org/api_docs/python/tf/estimator/train_and_evaluate))

Reply:
yeah, good point, I'm discussing this with Jianxun https://github.com/Microsoft/Recommenders/issues/539 He and I will work on this in the next release